10 video

Claudia108 opened the pull request:
These are the latest changes:
- add videos table
- add video link embed method in video model and tests
- add functionality to add, play, update and delete youtube videos 
- add controller tests for sites and videos.

Request for input: (I added the error messages inline)
There are 3 failures in my videos controller test: create (does increment count), update (does not find id for show path), delete (does not decrement count).
Same errors are in the Site controller test for create and update.
The error has to be in my tests because the feature tests pass and the web app works fine.
Any input is highly appreciated.

Claudia108 committed:
Adjust default image size

Claudia108 committed:
Add videos table, model and admin videos controller

Claudia108 committed:
Integrate video feature

Implement video index, show, add, edit/update and delete actions

Claudia108 committed:
Add controller tests for site and video

Claudia108 commented on test/controllers/admin/sites_controller_test.rb:
```diff
@@ -1,12 +1,40 @@
+    assert_equal 'no description available', assigns(:site).trinominal
+    assert_redirected_to site_path(assigns(:site))
+  end
+
```
Admin::SitesControllerTest#test_should_create_site [/Users/claudia/Turing/3module/projects/TexArch/test/controllers/admin/sites_controller_test.rb:14]:
"Site.count" didn't change by 1.
Expected: 4
  Actual: 3

Claudia108 commented on test/controllers/admin/sites_controller_test.rb:
```diff
@@ -1,12 +1,40 @@
-    assert site.has_attribute?("longitude")
+    assert_redirected_to site_path(assigns(:site))
   end
 end
```
2) Error:
Admin::SitesControllerTest#test_should_update_site:
ActionController::UrlGenerationError: No route matches {:action=>"show", :controller=>"sites", :id=>nil} missing required keys: [:id]
    test/controllers/admin/sites_controller_test.rb:38:in `block in class:SitesControllerTest'

Claudia108 committed:
Update test files

Claudia108 commented on test/controllers/admin/videos_controller_test.rb:
```diff
@@ -0,0 +1,44 @@
+    end
+    assert_redirected_to video_path(assigns(:video))
+  end
+
```
4) Failure:
Admin::VideosControllerTest#test_should_create_video [/Users/claudia/Turing/3module/projects/TexArch/test/controllers/admin/videos_controller_test.rb:14]:
"Video.count" didn't change by 1.
Expected: 3
  Actual: 2

Claudia108 commented on test/controllers/admin/videos_controller_test.rb:
```diff
@@ -0,0 +1,44 @@
+    end
+    assert_redirected_to videos_path
+  end
+end
```
3) Failure:
Admin::VideosControllerTest#test_should_destroy_video [/Users/claudia/Turing/3module/projects/TexArch/test/controllers/admin/videos_controller_test.rb:39]:
"Video.count" didn't change by -1.
Expected: 1
  Actual: 2

Claudia108 commented on test/controllers/admin/videos_controller_test.rb:
```diff
@@ -0,0 +1,44 @@
+    assert_response :success
+    assert_redirected_to video_path(assigns(:video))
+  end
+
```
5) Error:
Admin::VideosControllerTest#test_should_update_video:
ActionController::UrlGenerationError: No route matches {:action=>"show", :controller=>"videos", :id=>nil} missing required keys: [:id]
    test/controllers/admin/videos_controller_test.rb:34:in `block in class:VideosControllerTest'

jwashke commented:
Hey Claudia, I'm trying to get this running locally to take a look at it, but I'm missing some environment variables, and don't know which ones I need. Might be something to add to the readme.

Claudia108 commented:
Interesting. I have multiple keys for apis and paper clip in my
application.yml. Do you think this is what you are missing?
Thank you for taking a look!

On Fri, Jul 29, 2016 at 8:59 PM, Josh Washke notifications@github.com
wrote:

> Hey Claudia, I'm trying to get this running locally to take a look at it,
> but I'm missing some environment variables, and don't know which ones I
> need. Might be something to add to the readme.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Claudia108/TexArch/pull/62#issuecomment-236336412,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AOxIxBJ53zY7xmnczOduO3JCZDK0F7Jhks5qar4mgaJpZM4JYtD0
> .

jwashke commented:
Yeah if you could post the name of the variables in your application.yml that would be cool. You don't need to post the keys, I have an AWS account and what not but I need to know what keys I need and what they should be called in your application.

Claudia108 commented:
Ok, get it. these are all the variables from my application.yml:

SECRET_KEY_BASE

// for google oauth:

GOOGLE_CLIENT_ID
GOOGLE_CLIENT_SECRET

// for googlemaps.api:

google_api_key

// for paperclip:

aws_bucket: "texarch"
aws_access_key_id
aws_secret_access_key

I hope this is helpful. Thank you so much for doing this!

On Fri, Jul 29, 2016 at 9:18 PM, Josh Washke notifications@github.com
wrote:

> Yeah if you could post the name of the variables in your application.yml
> that would be cool. You don't need to post the keys, I have an AWS account
> and what not but I need to know what keys I need and what they should be
> called in your application.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Claudia108/TexArch/pull/62#issuecomment-236337542,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AOxIxFOvSzoQeuSOA_l3AA5wDDdWq0pmks5qasKigaJpZM4JYtD0
> .

jwashke commented:
Got it up and running this morning. A couple notes: 
- Migrating the database had an error, it looks like you define a name column in sites twice in your migration files.
  In create_sites

``` ruby
class CreateSites < ActiveRecord::Migration
  def change
    create_table :sites do |t|
      t.string :longitude
      t.string :latitude
      t.string :name
      t.string :type, default: "public"

      t.timestamps null: false
    end
  end
end
```

And in add name to site

``` ruby
class AddNameToSite < ActiveRecord::Migration
  def change
     add_column :sites, :name, :string
  end
end
```

I had to comment out the second add_column line to get the migrations to run

jwashke commented:
For the test errors themselves, I believe its because your not stubbing an admin login as all the errors are admin controller tests.

I commented out the line 
before_action :require_admin in the admin base controller and was left with these errors

```
  1) Failure:
Admin::VideosControllerTest#test_should_update_video [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/videos_controller_test.rb:33]:
Expected response to be a <success>, but was <302>


  2) Failure:
Admin::SitesControllerTest#test_should_create_site [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/sites_controller_test.rb:22]:
Expected: "no description available"
  Actual: "no trinominal available"


  3) Failure:
Admin::SitesControllerTest#test_should_update_site [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/sites_controller_test.rb:37]:
Expected response to be a <success>, but was <302>
```

Error number 1 and 3 I believe should be 302, you have assert_response :success, but success is for the status code 200 ok. Since you are redirecting in these paths (and are asserting you are redirecting you probably want to assert_response 302 or assert_response :redirect

You can read more about that [here](http://apidock.com/rails/ActionController/Assertions/ResponseAssertions/assert_response)

The last error, number 2 is from these two lines

``` ruby
    assert_equal 'no description available', assigns(:site).description
    assert_equal 'no description available', assigns(:site).trinominal
```

I'm wondering if this is just a copy-pasta error and am assuming the second line should be

``` ruby
    assert_equal 'no trinominal available', assigns(:site).trinominal
```

Since in your migration file that is the default line for the trinominal.

When I commented out the require admin before action, changed the two responses to assert_response :redirect, and changed the no description to no trinominal, the tests all passed. So I think the main issue is needing to stub an admin login for the tests. 

If thats not right let me know and I'm happy to take another look. Or let me know if you have further questions or anything else you want me to take a look at.

Claudia108 commented:
Wow, I am impressed. Thank you so much for your time and figuring out these
details. I will try out your suggestions and let you know how it goes.

Claudia

On Sat, Jul 30, 2016 at 11:10 AM, Josh Washke notifications@github.com
wrote:

> For the test errors themselves, I believe its because your not stubbing an
> admin login as all the errors are admin controller tests.
> 
> I commented out the line
> before_action :require_admin in the admin base controller and was left
> with these errors
> 
>   1) Failure:
> Admin::VideosControllerTest#test_should_update_video [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/videos_controller_test.rb:33]:
> Expected response to be a <success>, but was <302>
> 
>   2) Failure:
> Admin::SitesControllerTest#test_should_create_site [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/sites_controller_test.rb:22]:
> Expected: "no description available"
>   Actual: "no trinominal available"
> 
>   3) Failure:
> Admin::SitesControllerTest#test_should_update_site [/Users/Josh/turing/4_module/TexArch/test/controllers/admin/sites_controller_test.rb:37]:
> Expected response to be a <success>, but was <302>
> 
> Error number 1 and 3 I believe should be 302, you have assert_response
> :success, but success is for the status code 200 ok. Since you are
> redirecting in these paths (and are asserting you are redirecting you
> probably want to assert_response 302 or assert_response :redirect
> 
> You can read more about that here
> http://apidock.com/rails/ActionController/Assertions/ResponseAssertions/assert_response
> 
> The last error, number 2 is from these two lines
> 
> ```
> assert_equal 'no description available', assigns(:site).description
> assert_equal 'no description available', assigns(:site).trinominal
> ```
> 
> I'm wondering if this is just a copy-pasta error and am assuming the
> second line should be
> 
> ```
> assert_equal 'no trinominal available', assigns(:site).trinominal
> ```
> 
> Since in your migration file that is the default line for the trinominal.
> 
> When I commented out the require admin before action, changed the two
> responses to assert_response :redirect, and changed the no description to
> no trinominal, the tests all passed. So I think the main issue is needing
> to stub an admin login for the tests.
> 
> If thats not right let me know and I'm happy to take another look. Or let
> me know if you have further questions or anything else you want me to take
> a look at.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/Claudia108/TexArch/pull/62#issuecomment-236376998,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AOxIxDMIY1Tq18c48l2eeit2D4FZxlOAks5qa4WUgaJpZM4JYtD0
> .

jwashke commented:
Happy to help, let me know how it all worked out.

Claudia108 committed:
Add admin stub to failing tests

Claudia108 committed:
Add artifacts controller tests

Claudia108 commented:
Hi Josh, 
yes, you were right. I had to stub out the admin in the controller tests. Thank you for taking the time and for the suggestions. They all pass now. Just pushed up the changes. 
Do you know what how I could best fix the doubling migration? 
That must have happened in one of these very tired moments...
